addons/html_builder/static/tests/block_tab/snippet_groups.test.js

```diff
@@ -9,16 +9,16 @@ import {
     waitFor,
 } from "@odoo/hoot-dom";
 import { contains, onRpc } from "@web/../tests/web_test_helpers";
-import { defineWebsiteModels, setupWebsiteBuilder } from "../helpers";
+import { defineWebsiteModels, getSnippetStructure, setupWebsiteBuilder } from "../helpers";
 
 defineWebsiteModels();
 
-function getSnippetStructure({ name, content, keywords = [], groupName, imagePreview = "" }) {
-    keywords = keywords.join(", ");
-    return `<div name="${name}" data-oe-snippet-id="123" data-o-image-preview="${imagePreview}" data-oe-keywords="${keywords}" data-o-group="${groupName}">${content}</div>`;
-}
-
-function getBasicSection(content) {
+function getBasicSection(content, name, processed = false) {
```

Comment:
1) Another reason why not to have some processing changing the template, so we don't have "convoluted" checks.
2) If it is really necessary, you could always return the "processed" to avoid to having to write `getBasicSection(..., false)` and `getBasicSection(..., true)`.
3) Just a side note and detail, I would always avoid boolean arguments and have it wrapped in an object instead. Some arguments can be found here: https://alexkondov.com/should-you-pass-boolean-to-functions/